packages/remirror__react-core/src/hooks/use-extension.ts

```diff
@@ -90,8 +88,8 @@ export function useExtension<Type extends AnyExtensionConstructor>(
 ): void;
 export function useExtension<Type extends AnyExtensionConstructor>(
   Constructor: Type,
-  method: UseExtensionCallback<Type>,
-  dependencies: DependencyList,
+  memoizedCallback: UseExtensionCallback<Type>,
+  dependencies?: DependencyList,
```

Comment:
Seems like making this optional, (which it already is line 101) would fix the TypeScript hints for `addHandler`

Reply:
Below is the issue I see when using `useExtension` without the third argument:

<img width="1211" alt="image" src="https://user-images.githubusercontent.com/24715727/165222793-756a00c1-c694-47e2-ad16-fa9220695953.png">

Reply:
Below is my understanding of why `useExtension` with two arguments won't work, which may be incorrect.

`useExtension` has three overloads, shown as below:

```tsx
function useExtension(Constructor); // First overload
function useExtension(Constructor, options: DynamicOptionsOfConstructor); // Second overload
function useExtension(Constructor, callback: UseExtensionCallback, dependencies?); // Third overload
function useExtension(Constructor, optionsOrCallback = undefined, dependencies = []) {
  // actual implementation
}
```

When you pass three arguments to `useExtension`, TS will know that you are using the third overload since it's the only overload that can take three arguments, so that it can know that the type of the second argument is `UseExtensionCallback`. 

However, if you only pass two arguments, then TS won't know if the second argument is `DynamicOptionsOfConstructor` or `UseExtensionCallback`, so it won't be able to give the type of `addHandler`.

Reply:
> Here is the issue I see when using `useExtension` without the third argument:
> 
> <img alt="image" width="1211" src="https://user-images.githubusercontent.com/24715727/165222793-756a00c1-c694-47e2-ad16-fa9220695953.png">

Ah yes, I wasn't using `useCallback` .

We could fix the above by adding

```ts
export function useExtension<Type extends AnyExtensionConstructor>(
  Constructor: Type,
  method: (callback: UseExtensionCallback<Type>, deps: DependencyList) => UseExtensionCallback<Type>
): void;
```

Reply:
Interesting. Will try it.

Reply:
I fixed it! By switching the second and third overload. I guess this will make TS check the second overload (which is `UseExtensionCallback` now) sooner and everything works fine.

 
<img width="813" alt="image" src="https://user-images.githubusercontent.com/24715727/165293539-e6509673-3a30-4c73-9181-1efb74416793.png">